thorlcr/slave/traceslave.hpp

```diff
@@ -0,0 +1,58 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef TRACESLAVE_HPP
+#define TRACESLAVE_HPP
+
+#include "slave.hpp"
+
+class CTracingThorDataLink : implements CInterface, public IThorDataLink
+{
+private:
+    Owned<IThorDataLink> thorDataLink;
```

Comment:
Given this is per row, per activity, it would save a few cycles if methods below accessed a IThorDataLink pointer directly, versus indirectly through an Owned object operator and member ptr.
(and release in destructor)